Dockerfile

```diff
@@ -23,51 +22,17 @@ USER dev
 WORKDIR /home/dev
 
 # ===================================================================
-FROM base as pyenv
-
-RUN curl https://pyenv.run | bash
-RUN <<EOT bash
-    echo 'export PYENV_ROOT="\$HOME/.pyenv"' >> ~/.bashrc
-    echo 'command -v pyenv >/dev/null || export PATH="\$PYENV_ROOT/bin:\$PATH"' >> ~/.bashrc
-    echo 'eval "\$(pyenv init -)"' >> ~/.bashrc
-    echo 'export PYENV_ROOT="\$HOME/.pyenv"' >> ~/.profile
-    echo 'command -v pyenv >/dev/null || export PATH="\$PYENV_ROOT/bin:\$PATH"' >> ~/.profile
-    echo 'eval "\$(pyenv init -)"' >> ~/.profile
-EOT
-
-# ===================================================================
-FROM pyenv as py38
-RUN pyenv install 3.8
+FROM base as rye
+ARG PYTHON_VERSION=3.11
+RUN curl -sSf https://rye.astral.sh/get | RYE_INSTALL_OPTION="--yes" bash &&\
+    rye pin ${PYTHON_VERSION}
```

Comment:
This failed for me because the binary isn't on `PATH`. I solved this by either running `source "$HOME/.rye/env"` before the `rye` command as per the error output OR moving the `ENV PATH` statement (with the fix I noted) above this `RUN line.